addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -38,12 +38,15 @@ $('.oe_website_sale').each(function() {
     $(oe_website_sale).on('change', 'ul[data-attribute_value_ids]', function(event) {
         var $ul = $(event.target).closest('.js_add_cart_variants');
         var $parent = $ul.closest('.js_product');
-        var variant_ids = JSON.parse($ul.data("attribute_value_ids").replace(/'/g, '"'));
+        var variant_ids = $ul.data("attribute_value_ids");
+        if(_.isString(variant_ids)) {
+        	variant_ids = JSON.parse(variant_ids.replace(/'/g, '"'));
+        }
         var values = [];
         $parent.find('input.js_variant_change:checked, select.js_variant_change').each(function() {
             values.push(+$(this).val());
         });
-        var qty = $parent.find('input[name="add_qty"]').val();
+        var qty = parseInt($parent.find('input[name="add_qty"]').val() || 1);
```

Comment:
Should be fixed by https://github.com/odoo/odoo/pull/19970/commits/64b98c944efae9fcfc051f872a370dc34041786d

Reply:
Is it wrong to have the same data types in order to compare?